src/frontend/parallelization_phase.nit

```diff
@@ -66,15 +65,38 @@ private class ParallelizationPhase
 			classname += nmethdef.n_methid.as(AIdMethid).n_id.text
 		end
 
-		# Create a string corresponding to the threaded class
-		var s ="""
+		# String corresponding to the generated class
+		var s : nullable String = null
+
+		# case if the method has parameters
+		if nmethdef.n_signature.n_params.length != 0 then
+			var params = ""
+			for param in nmethdef.n_signature.n_params do
+				params += """
+var {{{param.n_id.text}}} : {{{param.n_type.n_id.text}}}
+
+"""
+			end
+			s="""
 class {{{classname}}}
 	super Thread
 
+	{{{params}}}
 	redef fun main do
 	end
 end
 """
+		else
+			s ="""
+class {{{classname}}}
+	super Thread
+
+	redef fun main do
+	end
+end
+"""
+		end
```

Comment:
This looks like code duplication. You could use the same super string (the first) and in the else case leave `params` an empty string (or Array).